core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -173,6 +181,24 @@ object UnitAutomation {
         if (unitDistanceToTiles.isEmpty()) return true // can't move, so...
         val currentUnitTile = unit.getTile()
 
+        // I'm sure this doesn't kill performance /s
+        val tilesWithRangedEnemyUnits = unit.currentTile.getTilesInDistance(3)
+                .map { it.getUnits().filter { unit.civInfo.isAtWarWith(it.civInfo) }.toList() }
```

Comment:
There's no reason to run .toList() here if you're going to flatten it later, I think.

Reply:
Actually better than that, use .flatMap() instead of .map()